### DISCUSSION TITLE
MongoDB recommended?

### DISCUSSION THREAD

**Discussion by slaudel** [ether/etherpad-lite#5706] [Q&A]
Hi,

as I'm rolling out a new instance of etherpad lite, I'm trying to switch away from MySQL/MariaDB to MongoDB (or any other no sql database). Unfortunately there is not too much documentation on using etherpad lite with databases other than MySQL.
After all it seems, I found how to configure it. 

But there are some deprecation warnings/errors thrown by etherpad lite about MongoDB like the following:

> [2023-03-31 17:55:50.776] [ERROR] console - (node:25051) [MONGODB DRIVER] Warning: Current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
(Use `node --trace-warnings ...` to show where the warning was created)
> [2023-03-31 17:55:50.786] [ERROR] console - (node:25051) [MONGODB DRIVER] Warning: Warning: no saslprep library specified. Passwords will not be sanitized
...
> [2023-03-31 17:58:48.378] [ERROR] console - (node:25051) DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.

So my question is, should I use MongoDB or should I stick with some relational database like MariaDB?

Kind regards
Sven

**Comment by webzwo0i** [comment]
I must admit I'm not using mongoDB.

I assume ueberdb's dependency on mongodb needs to be bumped (including handling any upcoming issues) to fix this. Someone with more knowledge needs to investigate what useUnifiedTopology and saslprep is about. I'd assume https://github.com/ether/ueberDB/blob/main/databases/mongodb_db.js#L100 can be changed to `updateOne`.

Currently resources for developing and maintaining Etherpad are very low. So I'd recommend you stick with MariaDB or Postgres. The database schema is extremely simple, so it is easy to switch database backends later.

**Comment by slaudel** [comment]
I'll stick with mariadb.
